### DISCUSSION TITLE
Configurable Product Option Order Fix

### DISCUSSION THREAD

**Discussion by Danieleeffe1** [OpenMage/magento-lts#3379] [Ideas]
With Magento 1.9x there was a bug in the order of the attributes.
when the list of attribute options is very long it was not possible to put the options in the desired order.

in openmage this bug has been fixed?

if not I post here a plug-in that fixes this error.
[Magento Configurable Product Option Order Fix.zip](https://github.com/OpenMage/magento-lts/files/12047921/Magento.Configurable.Product.Option.Order.Fix.zip)

**Comment by addison74** [comment]
Thank you for the post. 

This bug must be checked if it still exists in OpenMage, and if it exists then the offered solution required a PR. basically there is only one file to edit.

**Comment by fballiano** [comment]
the problem was in the backend? how many custom options are triggering the error? maybe it's just becuase of the many input fields in the form that are not submitted? in that case it can be fixed by setting the max_input_vars in your php.ini

https://stackoverflow.com/questions/10303714/php-max-input-vars

**Comment by Danieleeffe1** [comment]
1?) yes in backand.
2?) in my case it's more than 250 options which I sorted into groups by giving many options the same sort number.
3?) I found this patch years ago and it works great for me. I didn't have access to the php.ini at that time

**Comment by addison74** [comment]
I think that change refers rather to the position in the list of options of an attribute. I have not checked, but it is possible that the set position is not taken into account in the Frontend. Honestly speaking, I have never used it, but it is worth a test.

Regarding the Custom Option, which is a tab in the product edit page, I did not find any problems. Setting the position is corrected listed in Frontend.

![screenshot](https://github.com/OpenMage/magento-lts/assets/8360474/06dfbec8-9cad-40b0-a71a-e95b89d52e98)

**Reply by addison74** [reply]
If we take a look in the posted archive, we find that it is a matter of modifying a single file in the path /Block/Catalog/Product/View/Type/Configurable.php. 

There is a file in our repository /app/code/core/Mage/Catalog/Block/Product/View/Type/Configurable.php

**Reply by Danieleeffe1** [reply]
in my case the problem presented itself with a considerable number of options

these options were divided into ideal groups e.g.:

attribute name INGREDIENTS:

in the ingredients there are options:
spices ordered with the number 0: salt, pepper, rosemary, sage
cereals ordered with number 1: rice, durum wheat, soft wheat...
cheeses ordered with number 2: mozzrella, grana, hemmenthal...

and so on until reaching a considerable number of options (250) divided into 8 "groups"

**Reply by kiatng** [reply]
Why not have multiple smaller attributes: `spices`, `cereals`, `cheeses` instead of one big attribute `ingredients` containing 3 sub-attributes?

**Reply by Danieleeffe1** [reply]
In my specific case having multiple attributes is not practical.

I have to call out the ingredients in the frontend, both on the product page and in the product listing.

dividing it into several attributes and then calling them in the frontend would take up a huge amount of visual real estate
